mlir/tools/mlir-clang/Test/gemm.c

```diff
@@ -23,9 +23,15 @@ int main(void) {
   return 0;
 }
 
+// CHECK: #map0 = affine_map<(d0) -> (d0)>
+
 // CHECK: affine.for %arg0 = 0 to 1024 {
 // CHECK-NEXT: %{{.*}} = index_cast %arg0 : index to i32
 // CHECK-NEXT: affine.for %arg1 = 0 to 1024 {
 // CHECK-NEXT: %{{.*}} = index_cast %arg1 : index to i32
 // CHECK-NEXT: affine.for %arg2 = 0 to 1024 {
 // CHECK-NEXT: %{{.*}} = index_cast %arg2 : index to i32 
+
+// CHECK: affine.apply #map0(%arg0)
+// CHECK-NEXT: affine.apply #map0(%arg1)
+// CHECK-NEXT: affine.store %{{.*}}, %{{.*}}[%{{.*}}, %{{.*}}] : memref<?x1024xf32>
```

Comment:
Can you replace the regex with something that verifies the right index is being used? E.g. [[i:.+]]